File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.java
class Solution {
    public String reverseVowels(String s) {
        String vowels="aeiouAEIOU";
        int l=0,r=s.length()-1;
        char[] c=s.toCharArray();
        while(l<r){
            if(vowels.indexOf(c[l])==-1){
                l++;
            }
            else if(vowels.indexOf(c[r])==-1){
                r--;
            }
            else{
                char temp=c[l];
                c[l]=c[r];
                c[r]=temp;
                l++;
                r--;
            }
        }
        return new String(c);
    }
}

File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        vowels="aeiouAEIOU"
        l,r=0,len(s)-1
        c=list(s)
        while l<r:
            if c[l] not in vowels:
                l+=1
            elif c[r] not in vowels:
                r-=1
            else:
                c[l],c[r]=c[r],c[l]
                l+=1
                r-=1
        return "".join(c)
